pandas/core/groupby/groupby.py

```diff
@@ -487,10 +496,10 @@ def __init__(
         obj: FrameOrSeries,
         keys: Optional[_KeysArgType] = None,
         axis: int = 0,
-        level=None,
+        level: Optional[IndexLabel] = None,
```

Comment:
`IndexLabel` is likely too permissive here - would this ever be a scalar?

Reply:
From the groupby docs: level : int, level name, or sequence of such, default None

Reply:
E.g. 

```
df = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]}).set_index('a')
print(df.groupby(level='a').sum())
```

gives 

```
   b
a   
1  4
2  5
3  6
```